docs/how-to/test-file-properties.rst

```diff
@@ -51,30 +51,30 @@ Example
             @pytest.fixture(scope='session')
             @dt.working_directory(__file__)
             def df():
-                pattern = r'../*.*'
-                paths = (p for p in pathlib.Path('.').glob(pattern) if p.is_file())
+                paths = (p for p in pathlib.Path('..').glob('*.*') if p.is_file())
                 properties_dict = (get_properties(p) for p in paths)
                 df = pd.DataFrame.from_records(properties_dict)
-                df.set_index(['path'], inplace=True)
+                df = df.set_index(['path'])
                 return df
 
 
             def test_filetype(df):
+                suffixes = {'.csv', '.txt'}
                 def csv_or_txt(x):  # <- Helper function.
                     suffix = pathlib.Path(x).suffix
-                    return suffix.lower() in {'.csv', '.txt'}
+                    return suffix.lower() in suffixes
 
-                msg = 'Must be CSV or TXT files.'
+                msg = f'File suffix must be one of {suffixes}'
                 dt.validate(df.index, csv_or_txt, msg=msg)
 
 
             def test_filename(df):
-                msg = 'Must be lowercase with no spaces.',
+                msg = 'Must be lowercase with no spaces.'
                 dt.validate.regex(df['name'], r'[a-z0-9_.\-]+', msg=msg)
```

Comment:
My intention with the regex was to demonstrate that file names could be checked for any arbitrary pattern (not to validate all legal file names with lower case letters). But I can see that the example is misleading--this goes to your comment about keeping the message in sync with the regex pattern. I think this is a fair point to make and changing this will improve the example.

This can be changed to:

```
def test_filename(df):
    def is_lower_case(x):  # <- Helper function.
        return x.islower()

    dt.validate(df['name'], is_lower_case)
```


A more concise way to write this is:

```
from operator import methodcaller

...

def test_filename(df):
    dt.validate(df['name'], methodcaller('islower'))
```

Although I think the helper-function example will be easier for new users to understand.

Reply:
Well, TIL about `methodcaller`, neat!

Then let me build onto that idea:

```python
def test_filename(df):
    dt.validate(df['name'], str.islower)
```

`str.islower` is a callable just like `methodcaller("islower")`: `str.islower("Wat") == False`. The second one being a method has the advantage of being callable on non-`str` objects if they support `islower()`, but I don't think that scenario will ever happen. Apart from that, `str.islower`, `methodcaller("islower")` and your `is_lower_case` helper should be completely functionally equivalent, with the first one being the most intuitive and natural to me.

Reply:
But you could make two examples out of this:

1. check if file name as such is legal, e.g. by OS restrictions or something like lowercase
2. check if file name conforms to some human-made pattern, like `YYYY-MM-DD_data-stuff.csv`, rejecting all non-ISO 8601 file names. For this, regex is perfect, and the two things are distinct tests IMHO.

Reply:
On another page I added almost this *exact* example just last week:

https://datatest.readthedocs.io/en/latest/how-to/file-names.html#includes-date